### DISCUSSION TITLE
HM-800 sendet keine Daten an openDTU

### DISCUSSION THREAD

**Discussion by blackohw** [tbnobody/OpenDTU#1804] [General]
Hi,
ich habe eine kurze Frage zur Verbindung von WR und openDTU. Ich bin auf dem Gebiet nicht sonderlich nützlich, deswegen habe ich eine Fertige OPEN DTU mit ESP32 auf Ebay gekauft. (07/23) Ich bin jetzt erst dazu gekommen alles aufzubauen und anzuschließen. Das openDTU USB Teil hat sich zügig einrichten lassen und mit dem WLAN verbunden. Der HM-800 leuchter sekündlich grün, was auch ein gutes Zeichen ist.
Leider zeigt die openDTU keine Werte an. steht alles auf 0. Ist das ein aufwendiges Problem? Bin kein Programmierer und hab auch keinen Lötkolben zur Hand . Die Firmware habe ich aktualisiert.
Ich habe durch viele Beiträge gelesen aber durch sehr wenig Erfahrung konnte ich leider nichts hilfreiches aus anderen Threads rauslesen.
![DTU](https://github.com/tbnobody/OpenDTU/assets/162018617/be5c461c-4685-4291-a312-5ef744f3e1b2)
![DTU2](https://github.com/tbnobody/OpenDTU/assets/162018617/f0072cca-6af6-49be-840f-3c11c05ee57d)

**Comment by svalcken** [comment]
Deine OpenDTU findet nicht den Funkchip.
Prüf mal, ob der richtig verkabelt ist und ob es dir richtig ist.
Es gibt 2 Chips, nRF24 und der CMT2300A, es gibt DTUs in der Kombination je nur einen der beiden Chips oder auch beide.

Entweder ist die openDTU falsch eingestellt oder Du hast eine Hardware mit nur CMT2300A oder Funkchip ist defekt.

Die CMT2300A ist für die HMS Serie, solltest Du diesen haben, wird es leider nichts mit der Verbindung zum HM Wechselrichter, hier brauchst Du den nRF24 Chip.
Status und Typ müssen grün sein, verbunden und Chip Typ müssen da stehen.

Unter Einstellung Hardware siehst Du die Pin Belegung.

Du kannst mal Testweise die Konfig für CMT einstellen, wenn da stand beides Grün wird, hast Du die falsche Variante oder eine mit beiden Chips (Belegung eventuell anders).

Hast Du die Beschreibung zur OpenDTU oder Bilder der Hardware (Boards)?

**Comment by svalcken** [comment]
Hier ein Bild zu meiner OpenDTU mit beiden Chips:

![IMG_0245](https://github.com/tbnobody/OpenDTU/assets/2708976/d87e9657-8b82-4f00-b572-cb7ebb586fa3)

**Comment by blackohw** [comment]
Danke für die Antwort. ich hab alle Profile bei Einstellungen-> Hardware ausprobiert ohne Ergebnis, außer , das bei einem Profil nix mehr zu sehen war :D.
ich habe das 3D Gehäuse mal geöffnet und Bilder gemacht. Keine Ahnung ob die was taugen.
![IMG_20240303_185326422](https://github.com/tbnobody/OpenDTU/assets/162018617/352e3839-29a7-4bc5-8620-274833341aa6)
![IMG_20240303_185311772](https://github.com/tbnobody/OpenDTU/assets/162018617/f89fd3f3-6e0c-4a4c-8d97-436d9237a202)
![IMG_20240303_185340647](https://github.com/tbnobody/OpenDTU/assets/162018617/17ce0872-c839-48ea-aeea-2dc0856b93e2)

**Comment by svalcken** [comment]
Nach den Bildern,
Konnte ich erstmal den Chip identifizieren.
Ist nrf24, das ist schon mal richtig.

Siehe https://www.az-delivery.de/products/3x-nrf24l01-mit-2-4-ghz

Passt auch dazu, das bei Änderung der Konfiguration das nicht funktioniert hat.

Schreib dir die einzelnen Kabel auf, auf welche Pins diese Verbunden sind.

Das konnte ich gerade auf den Bildern leider kaum nachvollziehen.

Hast du das geflashed? Es gibt auch mehre Versionen von OpenDTU, die auf unterschiedliche Hardware gehen. Das könnte unter anderem die Zuordnung der Pins auch beeinflussen.

Schau das mal durch.

Hier noch ein Pinout vom ESP32-WROOM-32E

https://docs.sunfounder.com/projects/esp32-starter-kit/en/latest/_images/esp32_pinout.jpg

Müsste die draufsicht von oben sein.
Durchnummerierung beginnt links oben, geht dann links runter, weiter rechts unten und geht dann rauf bis zum rechte obersten.

Hier die Pinbelegung vom Funkmodul:
https://components101.com/sites/default/files/component_pin/nRF24L01-Pinout.png

Gibt es mehre Konfigurationen mit NRF? Wenn ja diese durch probieren.

Und bei den Profilen wird auch die Pinbelegung im Webinterface angezeigt.
Vergleichen mit der Verkabelung.

Viel Erfolg!

**Comment by blackohw** [comment]
danke für die Antwort. Ich probiere heut mal mein Glück. :)

**Comment by blackohw** [comment]
Ok, schwarz und gelb waren vertauscht.
![image](https://github.com/tbnobody/OpenDTU/assets/162018617/fefcabeb-188f-4cf9-80aa-3f0ab489d3c1)

Verbindung ist nun, theoretisch da. Daten empfängt er immernoch nicht. 
Problem könnte jetzt mein "Flash-Versuch" sein. Welche Firmware gehört dazu? Kann ich das "resetten"?

**Comment by blackohw** [comment]
ok, jetzt war ich zu ungeduldig, habe auf Werkseinstellungen zurückgesetzt. Jetzt geht nichts mehr außer dem roten LED Lämpchen. Wie bekomm ich da wieder Firmware drauf und welche? :D

**Comment by blackohw** [comment]
So, alles wieder geFlasht, DTU geht wieder. Immernoch kein Signale vom WR.

**Reply by Omega13x** [reply]
Nach Sonnenuntergang kommt da auch nichts mehr. Der WR wird ja über die PV-Module mit Strom versorgt. Keine Sonne, kein WR. Dafür braucht er aber nachts keinen Strom. Aber in der Konfiguration für den NRF ist ja jetzt alles grün. Sollte Morgen früh funktionieren.

**Comment by svalcken** [comment]
So.

Dann ist erstmal das Problem gelöst.

Der FunkChip sollte nun gehen.

Ggf. Die Dämpfung anpassen. Je nach Entfernung muss das noch angepasst werden.

Antenne ist auch dran, richtig?

Wechselrichter ist Grundsätzlich nur erreichbar, wenn dieser mit Strom aus den PV Modulen versorgt wird. Der Wechselrichter muss also auch blinken. Rotes Blinken, wenn der Wechselstrom nicht angeschossen ist. Grünes Blinken, wenn Pv und Wechselstrom korrekt ist.

Wechselrichter muss also Blinken, sonst wird in der DTU auch nichts ankommen.

Du hast keine Zweite DTU? z.B.: Hoymiles DTU Lite oder Pro?

Nach meiner Erfahrung verbindet sich der Wechselrichter nur an eine DTU. Ich habe eine Originale DTU und wenn der Wechselrichter sich daran verbunden hat, hat sich der Wechselrichter auch nicht mehr a die OpenDTU verbunden.

Nochmal Seriennummer prüfen, dass die richtig ist.

Zur Info, Gridprofil einstellen geht nur mit Original DTU. Das gleiche gilt für die Firmware.
Diese muss beim Support auch angefragt werden, wird aber nur geupdated, wenn es wirklich nötig ist.

Limit einstellen gibt es Permanent und Temporär und kann die OpenDTU per Webinterface oder mqtt. Entweder als Relativ ( per Prozent vom Maximalwert des Wechselrichter) oder Absolut mit direktem Leistungswert als Maximum.

Permanent wird bei jedem Starten (Wiederkehr vom PV Strom) wieder eingestellt. Temporär wird nicht gespeichert, ist aber gültig, bis zum nächsten Start, auch Neustart des Wechselrichter.

Stück für Stück geht es weiter.

Nachdem die DTU nun vermutlich empfangsbereit war, hättest du eventuell den Wechselrichter neu starten müssen.
Da du noch kein Befehl dazu senden konntest, hier die Alternative Lösung:
1. Wechselspannung abziehen.
2. Solarmodule bedecken oder pro Anschluss ein Stecker ziehen. Wichtig nicht unter Last, das kann dazu führen, das die Stecker durch Lichtbogen schaden nehmen, daher erst die 1.
3. LED muss aus sein.
4. Ca 5. Min warten.
5. Solarmodule abdecken bzw. Stecker wieder anstecken
6. Wechselstrom wieder einstecken
7. LED müsste erst Rot blinken und dann grün.

Danach an der DTU nochmal prüfen.
Wenn ich nicht irre verbindet sich der Wechselrichter nur beim Neustart und behält dann die Verbindung.

Ein Neustart der OpenDTU ist im normalen Betrieb nur sehr selten nötig. Wenn Du auf die CPU rechts in der Liveansicht klickst, sollte auch der Wechselrichter genau angezeigt werden, weiter unter auch eine Modellnummer mit mehr als 6 stellen.
Daneben gibt es noch das Steckdosensymbol und hier müsste das Gridprofil angezeigt werden. Da sollte DE - DE_VDE4105_2018 und Version 2.0 zu finden sein, wenn nicht bitte bei Unterstützung die Entwicklung aufklappen und prüfen, ob da mehre Zeile HexDez zahlenkolone zu sehen ist, ca. 4,5 Zeilen bei größerer Auflösung, Handy sind das noch mehr Zeilen.
Fürs Grid Einstellen wird die Original DTU gebraucht. Entweder selbst durchführen oder Jemand anderes kommt mit einer DTU und stellt es für Dich ein.

**Comment by blackohw** [comment]
Danke für die ausführliche Hilfe, Die chips der DTU scheinen soweit zu funktionieren.Eine Antenne kann ich nicht konkret ausfindig machen. Habe jetzt auch bei schönem Wetter die PV nach deiner Anleitung neu gestartet. Lämpchen von rot auf grün umgesprungen, alles scheinbar i.o.
OpenDTU erhält weiterhin Null infos...sonderbar. Ich habe mir mal die Hoymiles DTU bestellt, hoffe die kommt morgen, dann werd ich sehen obs am WR, an der DTU , oder an mir liegt :D
![openDTU](https://github.com/tbnobody/OpenDTU/assets/162018617/a50feb0b-1961-43aa-9100-4167f2b91550)

**Reply by barnosch** [reply]
Die Seriennummer vom WR hast Du peinlich genau überprüft und eingetragen?

**Reply by tbnobody** [reply]
Und wichtig ist auch nochmal den IRQ pin zu prüfen. Der kann falsch angeschlossen sein auch wenn die SPI Verbindung zum Modul besteht.

**Reply by svalcken** [reply]
Dass der IRQ nicht richtig angeschlossen ist, würde ich fast ausschliessen. Die Verkabelung wurde ja bereits geprüft und vertauschte Kabel gefunden.
Danach hat der Empfänger ja schon mal reagiert. Natürlich kann es nicht ganz ausgeschlossen werden.

Die Antenne ist auf der Platine mit drauf.

Möglich wäre auch, dass der Empfänger ein Defekt hat.

Mit der Original DTU kann es geprüft werden.
Bei mir mit der S Variante klappt es leider nicht diese Paralell zu betreiben. Die erste DTU, die gefunden wird, wird genutzt. Die andere DTU geht leer aus.

**Comment by blackohw** [comment]
Danke für eure Hilfe. Hab jetzt die DTU von Hoymiles angeschlossen, eingerichtet und alles funktioniert auf anhieb. Welches Problem auch immer die openDTU hat, werde ich wohl nicht mehr rausfinden (wollen). Vielen Dank für eure freundliche und fachkundige Unterstützung.
